demo/fiber/skynet_test/src/main/java/com/example/Skynet.java

```diff
@@ -0,0 +1,113 @@
+package com.example;/*
+ * Copyright (c) 2021, 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.TimeUnit;
+
+@Fork(3)
+@Warmup(iterations = 5, time = 10)
+@Measurement(iterations = 10, time = 10)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Thread)
```

Comment:
Just a question:
Which scope, `Scope.Benchmark` or `Scope.Thread`, should be used here?

Reply:
I believe that in this case, Scope.Thread would be better. Because each thread needs to have its own instance of Channel, rather than sharing one instance which ensures that each thread has its own independent state during execution without being affected by other threads.

Reply:
> ecause each thread needs to have its own instance of Channel, ...

It looks the benchmark method `skynet()` always instantiates a new `Channel` as the below,
`Channel<Long> chan = new Channel<>();`

And `@State` should affect `@Param`, exactly `num` in this case.

Reply:
I apologize for the confusion. The `num` is not modified during the recursive process. Each recursive call uses an independent variable `subNum` to store the computed value, ensuring that each thread has its own copy of the parameter. Therefore, it is possible to share the `num`  among all the threads. By using the `Scope.Benchmark,` we can avoid the overhead of copying the parameter for each thread and improve efficiency.I appreciate your advice and would like to confirm the accuracy of my understanding. Please feel free to correct me if I have misunderstood anything.

Reply:
In addition, I also conducted comparative testing and found that when `num` is 10000, using `Scope.Thread` yields better results with shorter and more stable execution times. This is likely because in this scenario, each thread has its own copy of the parameter, avoiding potential race conditions and improving efficiency and performance. On the other hand, when using `Scope.Benchmark`, where the parameter is shared, there may be competition among threads, resulting in a slight decrease in performance.